appgate/syncer/git.py

```diff
@@ -1,108 +1,126 @@
-        )
-        self.repository.git.push(
-            "--set-upstream", self.repository.remote().name, self.branch
+            f"[git-operator] Pushing changes to {self.git_repo.remote().name}:{branch}"
         )
+        if not dry_run:
+            self.git_repo.git.push(
+                "--set-upstream", self.git_repo.remote().name, branch
+            )
 
-    def create_pull_request(self) -> None:
+    def create_pull_request(self, branch: str, dry_run: bool) -> None:
         pass
 
 
+def github_repo(ctx: GitOperatorContext, repository_path: Path) -> GitRepo:
+    token = ensure_env(GITHUB_TOKEN_ENV)
+    # Fine-grained token? make sure the user is oauth2
+    if token.startswith("github_pat_"):
+        username = "oauth2"
+    elif not ctx.git_username:
+        raise AppgateException("Unable to find github username.")
+    else:
+        username = ctx.git_username
+    repository = f"github.com:{ctx.git_repository}"
+    log.info(f"[git-operator] Initializing the git repository by cloning {repository}")
+    if repository_path.exists():
+        shutil.rmtree(repository_path)
+    if not GITHUB_DEPLOYMENT_KEY_PATH.exists():
+        raise AppgateException(
+            f"Unable to find deployment key {GITHUB_DEPLOYMENT_KEY_PATH}"
+        )
+    git_repo = Repo.clone_from(
+        f"git@{repository}",
+        repository_path,
+        env={
+            "GIT_SSH_COMMAND": f"ssh -i {GITHUB_DEPLOYMENT_KEY_PATH} -o IdentitiesOnly=yes"
+        },
+    )
+    log.info(f"[git-operator] Repository {repository} cloned")
+    return GitHubRepo(
+        username=username,
+        token=token,
+        repository_name=repository,
+        git_repo=git_repo,
+        base_branch=ctx.git_base_branch,
+        vendor=ctx.git_vendor,
+        repository_path=repository_path,
+    )
+
+
+@attrs(slots=True, frozen=True)
 class GitHubRepo(GitRepo):
-    def check_env_vars(self) -> None:
-        super().check_env_vars()
-        envs = ["GITHUB_REPOSITORY"]
-        for env in envs:
-            if not os.getenv(env):
-                raise EnvironmentVariableNotFoundException(env)
-
-    def create_pull_request(self) -> None:
-        token = os.getenv("GIT_TOKEN")
-        base_branch = os.getenv("GIT_BASE_BRANCH", "master")
-        repo = os.getenv("GITHUB_REPOSITORY")
-        title = f"Merge changes from {self.branch}"
+    username: str = attrib()
+    token: str = attrib()
 
+    def create_pull_request(self, branch: str, dry_run: bool) -> None:
+        title = f"Merge changes from {branch}"
         log.info(
-            f"[git-syncer] Creating pull request in GitHub from '{self.branch}' to '{base_branch}'"
+            f"[git-operator] Creating pull request in GitHub from '{branch}' to '{self.base_branch}'"
         )
-        gh = Github(f"{token}")
-        gh_repo = gh.get_repo(f"{repo}")
+        if dry_run:
+            return
+        gh = Github(f"{self.token}")
```

Comment:
Unnecessary f-string:
```suggestion
        gh = Github(self.token)
```